src/dist/failure_detector/failure_detector.cpp

```diff
@@ -417,12 +417,17 @@ bool failure_detector::end_ping_internal(::dsn::error_code err, const beacon_ack
         return true;
     }
 
+    /*
```

Comment:
```is_master''' is considered by replication_failure_detector (layer 2), instead of by this failure detector (layer 1, where there is only 1 fd server which is always the master). The field ```is_master``` is put here due to the fact that it is not easy for the 2nd layer to patch the beacon messages, and it should be ignored here.